### ISSUE TITLE
XmlPoke Should Use DtdProcessing.Prohibit

### ISSUE THREAD

**Issue by BenVillalobos** [dotnet/msbuild#5817]
See https://github.com/dotnet/msbuild/issues/5810 for a bit of context.

There's an inconsistency with `XmlPoke` and `XslTransformation`. The [docs on XmlReaderSettings.DtdProcessing](https://docs.microsoft.com/dotnet/api/system.xml.xmlreadersettings.dtdprocessing) state that the default value is `Prohibit`, which neither `XmlPoke` or XslTransformation` follow.

**Comment by vijaya-lakshmi-venkatraman** [comment]
HI,
If this issue is still available, I would like to take this up.
I am a newbie to MSBuild and any pointers would be really helpful.

Thanks

**Comment by BenVillalobos** [comment]
@vijaya-lakshmi-venkatraman hello and welcome! When trying to build MSBuild, be sure to run `build.cmd` on the command line first.

Check out areas within XmlPoke (like [this one](https://github.com/dotnet/msbuild/blob/main/src/Tasks/XmlPoke.cs#L106)) that have instances of using DtdProcessing and see what it is set to. All instances should be set to DtdProcessing.Prohibit.

You can do the same with [XslTransformation](https://github.com/dotnet/msbuild/blob/main/src/Tasks/XslTransformation.cs#L222) and [XmlPeek](https://github.com/dotnet/msbuild/blob/main/src/Tasks/XmlPeek.cs#L214)

Thanks for taking this on, and feel free to ask many questions!

**Comment by vijaya-lakshmi-venkatraman** [comment]
Thanks so much. I understand the required changes. 
In XMLPeek.cs I see that there is a condition before we set DtdProcessing value. 
Should this be changed too?
```
DtdProcessing = prohibitDtd ? DtdProcessing.Prohibit : DtdProcessing.Ignore
```

**Comment by BenVillalobos** [comment]
Good question. That bool is passed in from the property on line 71. I think we can just set that property to true by default and leave that condition the same as an opt-out. We _probably_ shouldn't have it there at all, but it's been there for 3+ years so.../cc:@rainersigwald

**Comment by vijaya-lakshmi-venkatraman** [comment]
OK, to summarize the changes:
1) Set the default value of the property to DtdProcessing.Prohibit (XMLPeek.cs)
2) Leave the condition as is (XMLPeek.cs)
3) Set the value of DtdProcessing to DtdProcessing.Prohibit in all other places (XMLPoke.cs & XMLTransformation.cs)

Is that right?

**Comment by BenVillalobos** [comment]
Yep! In cases like `DtdProcessing = prohibitDtd ? DtdProcessing.Prohibit : DtdProcessing.Ignore`, consider leaving the check there, but defaulting the value of `prohibitDtd` to true.

I'm sure when the PR gets made we'll have other thoughts about it, but it's a good start!

**Comment by vijaya-lakshmi-venkatraman** [comment]
Thanks I will get the PR ready.
